### PR TITLE
Update showcase/references to match patterns used elsewhere

### PR DESCRIPTION
**Pull request by jblebrun**

### CODE REVIEW

**Comment by jblebrun** [comment]
(Still running thousands of tests to verify flakiness)

**Comment by jblebrun** [comment]
Unfortunately, still having lots of problems with test stability.

**Comment by jblebrun** [comment]
Leaving the `Ignore` annotation on the tests for now, but it's useful to get these changes in, since the structure matches patterns we are using elsewhere.

File: javatests/arcs/showcase/references/Arcs.kt [hunk 1]
```diff
@@ -3,27 +3,68 @@
...
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withTimeout
 import kotlin.coroutines.EmptyCoroutineContext
 
+/** Container for WriteRecipe specific things */
+@ExperimentalCoroutinesApi
+class ArcsStorage(private val arcs: Arcs) {
+    private inline fun <T> run(crossinline block: suspend () -> T) = runBlocking {
+        withTimeout(15000) {
```
Comment (shaper) [inline javatests/arcs/showcase/references/Arcs.kt:35]: If this is just to get more accurate test error info, can we add a comment to make this clear? It wasn't obvious to me when I was reading tests recently. We should also perhaps standardize on a constant here and put it in Archron test utils somewhere so it's easy to change everywhere if needed and easy to see where we do it.

File: javatests/arcs/showcase/references/Arcs.kt [hunk 2]
```diff
@@ -35,116 +76,77 @@ class Arcs(
...
     }
 
-    fun put0(item: MyLevel0) {
-        runBlocking {
-            startIfNot()
-            arcHost.writer0.write(item)
-        }
+    suspend fun startArc(plan: Plan): Arc {
+        val arc = allocator.startArcForPlan(plan)
+        arc.waitForStart()
```
Comment (shaper) [inline javatests/arcs/showcase/references/Arcs.kt:121]: Just confirming, is this valid/sufficient now or do we still need to do the workaround with checking all particle handles?

File: javatests/arcs/showcase/references/Arcs.kt [hunk 3]
```diff
@@ -35,116 +76,77 @@ class Arcs(
...
-        return runBlocking {
-            startIfNot()
-            arcHost.reader1.read()
-        }
+    suspend fun stopArcForPlan(plan: Plan) {
+        // Start the arc so that we have an instance of an arc to call stop on. Note this means that
+        // calling stop on an unstarted Arc will result in the Arc being started and immediately
+        // stopped.
+        val allocatorArc = allocator.startArcForPlan(plan)
+        allocatorArc.stop()
```
Comment (shaper) [inline javatests/arcs/showcase/references/Arcs.kt:130]: What happens if we call stop but it's not fully started? See comments above.

I also wonder whether this logic is at risk due to arc lifecycle race condition risk per https://github.com/PolymerLabs/arcs/pull/5575

File: javatests/arcs/showcase/references/Arcs.kt [hunk 4]
```diff
@@ -3,27 +3,68 @@
...
 import arcs.sdk.android.storage.ServiceStoreFactory
 import arcs.sdk.android.storage.service.ConnectionFactory
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withTimeout
 import kotlin.coroutines.EmptyCoroutineContext
 
+/** Container for WriteRecipe specific things */
```
Comment (shaper) [inline javatests/arcs/showcase/references/Arcs.kt:31]: If the intent is that this is generated by Archron or used by some Archron common lib logic down the road, perhaps worth noting in the comment.

I could be off base here, but wanted to ask while reading.

File: javatests/arcs/showcase/references/Arcs.kt (hunk 3 above)
Comment (jblebrun) [reply javatests/arcs/showcase/references/Arcs.kt:130]: The best logic here would be to look up an existing Arc for a given ID, and then stop that if it exists. I could not find a way to get an `Arc` by ID.

File: javatests/arcs/showcase/references/Arcs.kt (hunk 2 above)
Comment (jblebrun) [reply javatests/arcs/showcase/references/Arcs.kt:121]: Right now, the particles are also checking for handles to be ready. b/158233725 is still open, so I assume this is still required.

File: javatests/arcs/showcase/references/Arcs.kt (hunk 3 above)
Comment (jblebrun) [reply javatests/arcs/showcase/references/Arcs.kt:130]: (There should be a waitForStart here)

Comment (jblebrun) [reply javatests/arcs/showcase/references/Arcs.kt:130]: Hmmm, on further investigation, it seems that `AbstractArcHost.stopArc` is synchronous, so I think I can just convert this all to a `allocator.stopArc(plan.arcId)` call.

Comment (jblebrun) [reply javatests/arcs/showcase/references/Arcs.kt:130]: (reminded me that we want to file a similar task for storage proxy listeners, as well!)

**Pull request merged**